File: 프로그래머스/2/178870. 연속된 부분 수열의 합/연속된 부분 수열의 합.py
def solution(sequence, k):
    total = 0
    for start in range(len(sequence)-1,-1,-1):  # 뒤에서부터 진행
        total += sequence[start]
        if total > k:
            total -= sequence.pop() # 맨 끝자리 불필요
        if total == k:
            # 앞의 숫자가 중복일시
            while start > 0 and sequence[start-1] == sequence[-1]:
                start-=1
                sequence.pop()
            return([start,len(sequence)-1])

File: 프로그래머스/2/178870. 연속된 부분 수열의 합/연속된 부분 수열의 합.js
function solution(sequence, k) {
  var answer = [];
  let total = 0;

  for (let start = sequence.length - 1; start >= 0; start--) {
    total += sequence[start];
    if (total > k) total -= sequence.pop();
    if (total === k) {
      while (
        start > 0 &&
        sequence[start - 1] === sequence[sequence.length - 1]
      ) {
        start--;
        sequence.pop();
      }
      return [start, sequence.length - 1];
    }
  }
  return answer;
}